From dfbed49993b985937dee66a3f4ca38df41abd4c4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Sat, 29 Mar 2025 02:54:41 +0100
Subject: [PATCH] Rename new-model action to new-window

GNOME Shell uses the "new-window" action to open new main windows for an application. Renaming that action to this common name allows application launchers to use it.
---
 gaphor/ui/greeter.py    | 2 +-
 gaphor/ui/mainwindow.py | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/gaphor/ui/greeter.py b/gaphor/ui/greeter.py
index 3337cfb494..f72c1e133a 100644
--- a/gaphor/ui/greeter.py
+++ b/gaphor/ui/greeter.py
@@ -154,7 +154,7 @@ def close(self):
             self.greeter.destroy()
             self.greeter = None
 
-    @action(name="app.new-model", shortcut="<Primary>n")
+    @action(name="app.new-window", shortcut="<Primary>n")
     def new_model(self):
         self.open()
 
diff --git a/gaphor/ui/mainwindow.py b/gaphor/ui/mainwindow.py
index fdb4e5ea21..4ed50a271d 100644
--- a/gaphor/ui/mainwindow.py
+++ b/gaphor/ui/mainwindow.py
@@ -44,7 +44,7 @@ def create_hamburger_model(export_menu, tools_menu):
     model = Gio.Menu.new()
 
     part = Gio.Menu.new()
-    part.append(gettext("New Model…"), "app.new-model")
+    part.append(gettext("New Model…"), "app.new-window")
     part.append(gettext("Open Model…"), "app.file-open")
     model.append_section(None, part)
 
